sycl/test/regression/cache_test.cpp

```diff
@@ -0,0 +1,159 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -I %sycl_include/../ %s -o %t.out -L %sycl_libs_dir -lze_loader
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <algorithm>
+#include <level_zero/ze_api.h>
+#include <stdio.h>
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/backend/level_zero.hpp>
+#include <CL/sycl/backend/opencl.hpp>
+
+/*
+  The purpose of this test is to verify the expected behvior for
+  creating/caching platform and device objects. The available platforms and
+  devices (returned from get_platforms() and get_devices()) should remain
+  consistent throughout the lifetime of the application. Any new platform or
+  device queried should be one that is a part of the original list of platforms
+  and devices generated from get_platforms() and get_devices(). This test will
+  verify that in the following ways:
+    - Query for a list of all of the available platforms and devices
+    - Create a Queue using a default device selector and verify that the device
+      and platform associated with the queue can be found in the original
+      devices and platforms lists.
+    - Create both a platform and a device object using a native handle, and
+      verify that they are in the original platform and device lists.
+  This test will fail if it tries to create a platform or device that was not
+  in the list of platforms and devices queried in the begininning.
+*/
+
+// Query the device and platform from a queue (constructed with a default device
+// selector) and check if they can be found in the devices and platforms lists
+// that were generated above.
+int queryFromQueue(std::vector<cl::sycl::platform> *platform_list,
+                   std::vector<cl::sycl::device> *device_list) {
+  int failures = 0;
+  cl::sycl::queue Q{cl::sycl::default_selector{}};
+  cl::sycl::device dev = Q.get_info<cl::sycl::info::queue::device>();
+  auto plt = dev.get_platform();
+
+  std::cout << "Platform queried from Queue : "
+            << plt.get_info<cl::sycl::info::platform::name>() << std::endl;
+  auto plt_result = std::find_if(
+      platform_list->begin(), platform_list->end(), [&](cl::sycl::platform &p) {
+        return p.get_info<cl::sycl::info::platform::name>() ==
+               plt.get_info<cl::sycl::info::platform::name>();
```

Comment:
You should be able to compare the platform objects for equality, not just their names.  Can you change this to:

```
return p == plt;
```

If that does not work, there is some other bug in the DPC++ runtime.

Reply:
Done.